File: runtime/java/units/table/call.java
// requires: rt/trap
// The type string is a structural key, not a module-local type index, so a table
// shared across modules stays consistent.
Rt.Fn call(int index, String ty) {
    if (index < 0 || index >= slots.length) {
        Rt.trap("undefined element");
    }
    Rt.Funcref f = slots[index];
    if (f == null) {
        Rt.trap("uninitialized element");
    }
    if (!f.ty.equals(ty)) {
        Rt.trap("indirect call type mismatch");
    }
    return f.fn;
}

File: crates/dewasm-backend-python/units/table/call.py
# requires: rt/trap
def call(self, i, type_key, *args):
    if i >= len(self._slots):
        Rt.trap("undefined element")
    slot = self._slots[i]
    if slot is None:
        Rt.trap("uninitialized element")
    if slot[0] != type_key:
        Rt.trap("indirect call type mismatch")
    return slot[1](*args)
